Model/Config/Source/CrossBorderClass/ProductAttributes.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * ClassyLlama_AvaTax
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @copyright  Copyright (c) 2018 Avalara, Inc.
+ * @license    http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ */
+
+namespace ClassyLlama\AvaTax\Model\Config\Source\CrossBorderClass;
+
+use Magento\Eav\Model\ResourceModel\Entity\Attribute\Collection;
+
+class ProductAttributes implements \Magento\Framework\Option\ArrayInterface
+{
+    /**
+     * @var \Magento\Eav\Model\ResourceModel\Entity\Attribute\CollectionFactory
+     */
+    protected $collectionFactory;
+
+    /**
+     * @param \Magento\Eav\Model\ResourceModel\Entity\Attribute\CollectionFactory $collectionFactory
+     */
+    public function __construct(\Magento\Eav\Model\ResourceModel\Entity\Attribute\CollectionFactory $collectionFactory)
+    {
+        $this->collectionFactory = $collectionFactory;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function toOptionArray($isMultiselect = true, $foregroundCountries = '')
+    {
+        /** @var Collection $collection */
+        $collection = $this->collectionFactory->create();
+        $collection->addFieldToSelect('attribute_code')
+            ->addFieldToSelect('frontend_label')
+            ->addFieldToFilter('backend_type', ['in' => ['int', 'varchar']])
+            ->addFieldToFilter('frontend_label', ['notnull' => true])
+            ->setEntityTypeFilter(4)
```

Comment:
Using magic numbers, I see. 
![image](https://user-images.githubusercontent.com/129031/44067144-951fb2b2-9f39-11e8-830b-0760b139d941.jpg)

This entity is not guaranteed to be "4". I'd do something like what is being done in `\Magento\Catalog\Model\Config\CatalogClone\Media\Image::getPrefixes`:

```
    public function getPrefixes()
    {
        // use cached eav config
        $entityTypeId = $this->_eavConfig->getEntityType(\Magento\Catalog\Model\Product::ENTITY)->getId();

        /* @var $collection \Magento\Catalog\Model\ResourceModel\Product\Attribute\Collection */
        $collection = $this->_attributeCollectionFactory->create();
        $collection->setEntityTypeFilter($entityTypeId);
        $collection->setFrontendInputTypeFilter('media_image');
        …
```